### PR TITLE
update rules for borrower app v2

### PR DESCRIPTION
**Pull request by spencersteers**
- eslint 8
- upgrade deps

I tested all these changes in server to make sure we are okay to upgrade

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -48,7 +47,8 @@ module.exports = {
           },
           Symbol: {
             message: 'Avoid using the `Symbol` type. Did you mean `symbol`?'
-          }
+          },
+          '{}': false
```
Comment (spencersteers) [inline index.js:51]: `{}` is the default type for react props

File: react.js [hunk 2]
```diff
@@ -49,15 +46,14 @@ module.exports = {
     "react/no-did-mount-set-state": "error",
     "react/no-did-update-set-state": "error",
     "react/no-find-dom-node": "off",
-    "react/no-multi-comp": "error",
     "react/no-unescaped-entities": "error",
     "react/self-closing-comp": "error",
-    "react/sort-comp": "error",
```
Comment (spencersteers) [inline react.js:55]: this conflicts with `@typescript-eslint/member-ordering`

File: react.js [hunk 3]
```diff
@@ -49,15 +46,14 @@ module.exports = {
     "react/no-did-mount-set-state": "error",
     "react/no-did-update-set-state": "error",
     "react/no-find-dom-node": "off",
-    "react/no-multi-comp": "error",
```
Comment (spencersteers) [inline react.js:52]: sometimes you just have a one liner component

File: react.js [hunk 4]
```diff
@@ -1,40 +1,37 @@
...
-  "rules": {
+  rules: {
     "jsx-quotes": ["error", "prefer-single"],
     "no-console": ["error", { allow: ["warn", "error"] }],
     "no-underscore-dangle": ["error", { "allow": ["_maxListeners", "_id"] }],
     "react/jsx-boolean-value": "error",
     "react/jsx-closing-bracket-location": ["error", "tag-aligned"],
     "react/jsx-curly-spacing": "error",
     "react/jsx-equals-spacing": "error",
-    "react/jsx-handler-names": "error",
```
Comment (spencersteers) [inline react.js:37]: sometimes the props you get are the props you get

File: index.js [hunk 5]
```diff
@@ -325,37 +325,6 @@ module.exports = {
...
       }
-    },
-    // TS shared overrides should be put into the shared eslint config
-    {
-      files: ['**/*.ts'],
-      rules: {
-        // This rule has to live here because it spits out random warning messages if run on JS files
-        // would be preferable if we could drop this rule since it uses tslint which we would like to drop
-        // completely
-        '@typescript-eslint/tslint/config': [
```
Comment (spencersteers) [inline index.js:336]: nothing to see here pretend you never saw this

File: mocha.js [hunk 6]
```diff
@@ -26,18 +26,14 @@ module.exports = {
         'mocha/max-top-level-suites': 'off',
         'mocha/no-mocha-arrows': 'off',
         // Off because chai types are bad and hits this in a lot of places
-        '@typescript-eslint/no-unused-expressions': 'off',
-        // Off because mocha this.timeout requires none arrow functions
-        'prefer-arrow/prefer-arrow-functions': 'off'
```
Comment (spencersteers) [inline mocha.js:31]: some lingering `prefer-arrow` rules that weren't removed

**Review by m-abboud — APPROVED** [review]
looks good

**Pull request merged**